src/o365/todo/commands/list/list-remove.ts

```diff
@@ -0,0 +1,166 @@
+import commands from '../../commands';
+import request from '../../../../request';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate, CommandError
+} from '../../../../Command';
+import GraphCommand from '../../../base/GraphCommand';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id?: string;
+  name?: string;
+  confirm?: boolean;
+}
+
+class TodoListRemoveCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.LIST_REMOVE}`;
+  }
+
+  public get description(): string {
+    return 'Removes a Microsoft To Do task list';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+
+    const getListId = () => {
+      if (args.options.name) {
+        // Search list by its name
+        const requestOptions: any = {
+          url: `${this.resource}/beta/me/todo/lists?$filter=displayName eq '${escape(args.options.name)}'`,
+          headers: {
+            accept: "application/json;odata.metadata=none"
+          },
+          json: true
+        };
+        return request.get(requestOptions)
+          .then((response: any) => response.value && response.value.length === 1 ? response.value[0].id : null);
+      }
+
+      return Promise.resolve(args.options.id);
+    };
+
+    const removeList = () => {
+      getListId().then(listId => {
+        if (!listId) {
+          cb(new CommandError(`The list ${args.options.name} cannot be found`));
+          return;
+        }
+
+        const requestOptions: any = {
+          url: `${this.resource}/beta/me/todo/lists/${listId}`,
+          headers: {
+            accept: "application/json;odata.metadata=none"
+          },
+          json: true
+        };
+
+        request
+          .delete(requestOptions)
+          .then((): void => {
+
+            if (this.verbose) {
+              cmd.log(vorpal.chalk.green('DONE'));
+            }
+
+            cb();
+          }, (err: any) => this.handleRejectedODataJsonPromise(err, cmd, cb));
```

Comment:
Instead of nesting promises and handling errors twice, we can simplify the code by changing line 64 to `return request.delete` and then handle the response just once.